client/src/app/index.js

```diff
@@ -1,13 +1,43 @@
 // npm packages
-import React from 'react';
+import React, {Component} from 'react';
+import {connect} from 'react-redux';
+import {push} from 'react-router-redux';
 
 import Header from '../components/header';
 import Footer from '../components/footer';
 
-export default ({children}) => (
-  <div className="container">
-    <Header />
-    {children}
-    <Footer />
-  </div>
-);
+const mapStateToProps = state => ({
+  actualPath: state.routing.locationBeforeTransitions && state.routing.locationBeforeTransitions.pathname,
+  user: state.auth.user,
+  token: state.auth.token,
+});
+
+const mapDispatchToProps = dispatch => ({
+  navTo: (path) => dispatch(push(path)),
+});
+
+class App extends Component {
```

Comment:
You're right, this is not a very good solution but I think it's better than previously. I think we should definitely use redux-auth-wrapper. Now I do not have time but I can investigate it later.